engine/src/test/scala/cromwell/engine/workflow/workflowstore/SqlWorkflowStoreSpec.scala

```diff
@@ -244,22 +253,128 @@ class SqlWorkflowStoreSpec extends AnyFlatSpec with CromwellTimeoutSpec with Mat
+
+        // there are 10 workflows from hog group "Zardoz" in the store, but since the group is excluded, 0 workflows are returned here
+        startableWorkflows3 <- workflowStore.fetchStartableWorkflows(5, "A08", 5.minutes, excludedGroups = Set("Zardoz"))
+        _ = startableWorkflows3.size should be(0)
+
+        // hog group "Zardoz" has tokens to run workflows, hence don't exclude it
+        startableWorkflows4 <- workflowStore.fetchStartableWorkflows(5, "A08", 5.minutes, Set.empty[String])
+        _ = startableWorkflows4.map(_.hogGroup.value).toSet.head should be("Zardoz")
+        _ = startableWorkflows4.map(_.id).foreach(x => zardozWorkflowIds.toList should contain(x))
+        _ = updateWfToRunning(startableWorkflows4)
+
+        startableWorkflows5 <- workflowStore.fetchStartableWorkflows(5, "A08", 5.minutes, Set.empty[String])
+        _ = startableWorkflows5.map(_.hogGroup.value).toSet.head should be("Zardoz")
+        _ = startableWorkflows5.map(_.id).foreach(x => zardozWorkflowIds.toList should contain(x))
+        _ = updateWfToRunning(startableWorkflows5)
+
+        // remove entries from WorkflowStore
+        workflowsList = goldFingerWorkflowIds ++ zardozWorkflowIds
+        _ = workflowsList.foreach(id => Await.result(workflowStore.deleteFromStore(id), 5.seconds))
+      } yield()).futureValue
+    }
+
+    it should "start workflows from hog group with lowest count of running workflows for multiple hog groups" taggedAs DbmsTest in {
       (for {
-        submissionResponsesExcluded <- workflowStore.add(excludedGroupSourceFilesCollection)
-        submissionResponsesIncluded1 <- workflowStore.add(includedGroupSourceFilesCollection1)
-        submissionResponsesIncluded2 <- workflowStore.add(includedGroupSourceFilesCollection2)
-        submissionResponsesIncluded3 <- workflowStore.add(includedGroupSourceFilesCollection3)
-        startableWorkflows <- workflowStore.fetchStartableWorkflows(3, "A08", 1.second, excludedGroups = Set("Zardoz"))
-        _ = startableWorkflows.map(_.id).intersect(submissionResponsesExcluded.map(_.id).toList).size should be(0)
-        _ = startableWorkflows.map(_.id).intersect(submissionResponsesIncluded1.map(_.id).toList).size should be(1)
-        _ = startableWorkflows.map(_.id).intersect(submissionResponsesIncluded2.map(_.id).toList).size should be(1)
-        _ = startableWorkflows.map(_.id).intersect(submissionResponsesIncluded3.map(_.id).toList).size should be(1)
-        _ = startableWorkflows.map(_.id).size should be(3)
-        _ <- workflowStore.deleteFromStore(submissionResponsesExcluded.head.id) // Tidy up
-        _ <- workflowStore.deleteFromStore(submissionResponsesIncluded1.head.id) // Tidy up
-        _ <- workflowStore.deleteFromStore(submissionResponsesIncluded2.head.id) // Tidy up
-        _ <- workflowStore.deleteFromStore(submissionResponsesIncluded3.head.id) // Tidy up
+        // first submission of 10 workflows for hogGroup "Goldfinger"
```

Comment:
To clarify, our logic is to only compare two hog groups at a time, and as one finishes, another starts (if there is another)?

Reply:
No. First we find out which hog group has the lowest count of actively running workflows among all the hog groups present in Workflow Store table. And then start new workflows from that hog group. This way Cromwell is fair in terms of starting workflows almost equally from various hog groups. Does that make sense?